flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -86,7 +84,7 @@
 
 	private boolean isDestroyed;
 
-	private BufferPoolOwner owner;
+	private final BufferPoolOwner owner;
```

Comment:
please change it to `private final Optional<BufferPoolOwner> owner`. This value is logically optional and should be marked as such (we can not use `@Nullable` because we are not enforcing nullable checks during compilation/checkstyle). 

Ditto for the constructor and method parameters that are providing this value. Either overload them by providing both:
```
BufferPool createBufferPool(int numRequiredBuffers, int maxUsedBuffers, BufferPoolOwner owner);

BufferPool createBufferPool(int numRequiredBuffers, int maxUsedBuffers);
```
or if you are lazy provide one method:
```
BufferPool createBufferPool(int numRequiredBuffers, int maxUsedBuffers, Optional<BufferPoolOwner> owner)
```
but the first option is more clean.

Reply:
Yes, the optional way is better, and I will take the first option as you suggested.